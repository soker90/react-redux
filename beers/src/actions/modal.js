import modalAction from '../constants/actions/modal'

export default {
  openModal: modalData => ({type: modalAction.OPEN_MODAL, modalData}),
  closeModal: () => ({type: modalAction.CLOSE_MODAL})
}
