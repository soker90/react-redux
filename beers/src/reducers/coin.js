import coinAction from '../constants/actions/coin'

export function initialState () {
  return {
    coinList: [],
    coin: {},
    loading: false,
    error: null
  }
}

function fetchCoin(state) {
  return {
    ...state,
    coinList: [],
    loading: true,
    error: null
  }
}

function fetchCoinSuccess(state, {coins}) {
  return {
    ...state,
    coinList: coins.coins,
    loading: false,
    error: null
  }
}

function addCoin(state) {
  return {
    ...state,
    loading: true,
    error: null
  }
}

function addCoinSuccess(state, {coin}) {
  return {
    ...state,
    coinList: [coin, ...state.coinList],
    loading: false,
    error: null
  }
}

function removeCoin(state) {
  return {
    ...state,
    loading: true,
    error: null
  }
}

function removeCoinSuccess(state, {coinId}) {
  return {
    ...state,
    coinList: state.coinList.filter((coin) => coin.id !== coinId.coinId),
    loading: false,
    error: null
  }
}

function coinError(state, {error}) {
  return {
    ...state,
    loading: false,
    error: error
  }
}

export default function (state = initialState(), action) {
  switch (action.type) {
    case coinAction.FETCH_COIN:
      return fetchCoin(state, action)
    case coinAction.FETCH_COIN_SUCCESS:
      return fetchCoinSuccess(state, action)
    case coinAction.ADD_COIN:
      return addCoin(state, action)
    case coinAction.ADD_COIN_SUCCESS:
      return addCoinSuccess(state, action)
    case coinAction.REMOVE_COIN:
      return removeCoin(state, action)
    case coinAction.REMOVE_COIN_SUCCESS:
      return removeCoinSuccess(state, action)
    case coinAction.COIN_ERROR:
      return coinError(state, action)
    default:
      return state
  }
}
