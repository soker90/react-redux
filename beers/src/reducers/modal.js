import modalAction from '../constants/actions/modal'

export function initialState () {
  return {
    open: false,
    title: '',
    body: '',
    callback: null,
    extraParams: {}
  }
}

function openModal(state, {modalData}) {
  return {
    ...state,
    open: true,
    title: modalData.title,
    body: modalData.body,
    callback: modalData.callback ? modalData.callback : null,
    extraParams: modalData.extraParams ? modalData.extraParams : {}
  }
}

function closeModal(state) {
  return {
    ...state,
    open: false,
    title: '',
    body: '',
    callback: null,
    extraParams: {}

  }
}

export default function (state = initialState(), action) {
  switch (action.type) {
    case modalAction.OPEN_MODAL:
      return openModal(state, action)
    case modalAction.CLOSE_MODAL:
      return closeModal(state, action)
    default:
      return state
  }
}
