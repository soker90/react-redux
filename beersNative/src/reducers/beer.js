import beerAction from '../constants/actions/beer'

export function initialState () {
  return {
    beerList: [],
    beer: {},
    loading: false,
    error: null
  }
}

function fetchBeer(state) {
  return {
    ...state,
    beerList: [],
    loading: true,
    error: null
  }
}

function fetchBeerSuccess(state, {beers}) {
  return {
    ...state,
    beerList: beers.beers,
    loading: false,
    error: null
  }
}

function addBeer(state) {
  return {
    ...state,
    loading: true,
    error: null
  }
}

function addBeerSuccess(state, {beer}) {
  return {
    ...state,
    beerList: [beer, ...state.beerList],
    loading: false,
    error: null
  }
}

function removeBeer(state) {
  return {
    ...state,
    loading: true,
    error: null
  }
}

function removeBeerSuccess(state, {beerId}) {
  return {
    ...state,
    beerList: state.beerList.filter((beer) => beer.id !== beerId.beerId),
    loading: false,
    error: null
  }
}

function beerError(state, {error}) {
  return {
    ...state,
    loading: false,
    error: error
  }
}

export default function (state = initialState(), action) {
  switch (action.type) {
    case beerAction.FETCH_BEER:
      return fetchBeer(state, action)
    case beerAction.FETCH_BEER_SUCCESS:
      return fetchBeerSuccess(state, action)
    case beerAction.ADD_BEER:
      return addBeer(state, action)
    case beerAction.ADD_BEER_SUCCESS:
      return addBeerSuccess(state, action)
    case beerAction.REMOVE_BEER:
      return removeBeer(state, action)
    case beerAction.REMOVE_BEER_SUCCESS:
      return removeBeerSuccess(state, action)
    case beerAction.BEER_ERROR:
      return beerError(state, action)
    default:
      return state
  }
}
