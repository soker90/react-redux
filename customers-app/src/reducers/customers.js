import { handleActions } from 'redux-actions'
import { FETCH_CUSTOMERS, INSERT_COSTUMER } from '../constants'

export const customers = handleActions({
	[FETCH_CUSTOMERS]: (state, action) => [...action.payload],
	[INSERT_COSTUMER]: (state, action) => [...state, action.payload],
}, [])

