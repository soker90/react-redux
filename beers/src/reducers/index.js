import { combineReducers } from 'redux'
import beers from './beer'
import modal from './modal'
import coins from './coin'
import { reducer as formReducer } from 'redux-form'

const rootReducer = combineReducers({
  beers,
  modal,
  coins,
  form: formReducer
})

export default rootReducer
