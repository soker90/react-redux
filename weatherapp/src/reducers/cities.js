import { GET_WEATHER_CITY, SET_FORECAST_DATA, SET_WEATHER_CITY } from '../actions'
import { createSelector } from 'reselect'
import toPairs from 'lodash.topairs'

export const cities = (state = {}, action) => {
	switch (action.type) {
		case SET_FORECAST_DATA:
			const {city, forecastData} = action.payload
			return {...state, [city]: {...state[city], forecastData, forecastDataDate: new Date()}}
		case GET_WEATHER_CITY: {
			const city = action.payload
			return {...state, [city]: {...state[city], weather: null}}
		}

		case SET_WEATHER_CITY: {
			const {city, weather} = action.payload
			return {...state, [city]: {...state[city], weather}}
		}
		default:
			return state
	}

}

export const getForecastDataFromCities = createSelector((state, city) => state[city] && state[city].forecastData, forecastData => forecastData)

const fromObjToArray = cities => (toPairs(cities).map(([key,value]) => ({ key, name: key, data: value.weather})))
export const getWeatherCities =
	createSelector(state => fromObjToArray(state), cities => cities)