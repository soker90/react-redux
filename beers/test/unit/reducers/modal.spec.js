// modal.spec.js
import modalAction from '../../../src/actions/modal'
import modalReducer from '../../../src/reducers/modal'
import { initialState } from '../../../src/reducers/modal'

describe('Modal reducer', () => {

  test('should handle initial state', () => {
    expect(initialState()).toMatchSnapshot()
  })

  test('openModal method should handle open modal and set params', () => {
    const result = modalReducer(initialState(), modalAction.openModal({
          title: 'Hey a modal title',
          body: 'This is the modal body text',
          callback: jest.fn()
      }))

    expect(result).toMatchSnapshot()
  })

  test('closeModal method should handle close modal and reset params', () => {
    const result = modalReducer(initialState(), modalAction.closeModal())

    expect(result).toMatchSnapshot()
  })

})
