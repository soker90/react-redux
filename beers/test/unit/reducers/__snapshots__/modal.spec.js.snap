exports[`Modal reducer closeModal method should handle close modal and reset params 1`] = `
Object {
  "body": "",
  "callback": null,
  "extraParams": Object {},
  "open": false,
  "title": "",
}
`;

exports[`Modal reducer openModal method should handle open modal and set params 1`] = `
Object {
  "body": "This is the modal body text",
  "callback": [Function],
  "extraParams": Object {},
  "open": true,
  "title": "Hey a modal title",
}
`;

exports[`Modal reducer should handle initial state 1`] = `
Object {
  "body": "",
  "callback": null,
  "extraParams": Object {},
  "open": false,
  "title": "",
}
`;
